devito/passes/clusters/cse.py

```diff
@@ -90,12 +90,13 @@ def _cse(maybe_exprs, make, min_cost=1, mode='default'):
 
     while True:
         # Detect redundancies
-        counted = count(processed).items()
-        targets = OrderedDict([(k, estimate_cost(k, True)) for k, v in counted if v > 1])
+        counted = count(processed, None).items()
+        targets = OrderedDict([(k, estimate_cost(k[0], True))
```

Comment:
why k[0] specifically?

Reply:
see the `count` output

but yes, we should use a namedtuple or a Bunch, see my other comments below